drivers/power/ltc3350/ltc3350.c

```diff
@@ -0,0 +1,371 @@
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+* LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+* SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+* CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+* OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <errno.h>
+
+#include "no_os_util.h"
+#include "no_os_delay.h"
+#include "no_os_alloc.h"
+#include "no_os_i2c.h"
+
+#include "ltc3350.h"
+
+/******************************************************************************/
+/************************ Variable Declarations ******************************/
+/******************************************************************************/
+
+/******************************************************************************/
+/************************ Functions Declarations ******************************/
+/******************************************************************************/
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/***************************************************************************//**
+
+/***************************************************************************//**
+ * @brief Initializes the communication peripheral and checks if the LTC3350
+ *        part is present.
+ *
+ * @param device     - The device structure.
+ * @param init_param - The structure that contains the device initial
+ * 		       parameters.
+ *
+ * @return ret       - Result of the initialization procedure.
+*******************************************************************************/
+int ltc3350_init(struct ltc3350_dev **device,
+		 struct ltc3350_init_param *init_param)
+{
+	struct ltc3350_dev *dev;
+	int ret = 0;
+	dev = (struct ltc3350_dev *)malloc(sizeof(*dev));
+	if (!dev)
+		return -ENOMEM;
+
+	ret = no_os_i2c_init(&dev->i2c_desc, &init_param->i2c_init);
+
+	if (ret) {
+		no_os_free(dev);
+	} else {
+		*device = dev;
+	}
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Free the resources allocated by ltc3350_init().
+ *
+ * @param dev - The device structure.
+ *
+ * @return ret - Result of the remove procedure.
+*******************************************************************************/
+int ltc3350_remove(struct ltc3350_dev *dev)
+{
+	int ret = 0;
```

Comment:
initializing ret with 0 here does not add much value. since it is initialized again in the next line of code.